arch/Kconfig

```diff
@@ -381,6 +381,22 @@ config EXTRA_EXCEPTION_INFO
 
 endmenu # Interrupt configuration
 
+config INIT_ARCH_HW_AT_BOOT
+	bool "Initialize Core Achitecture HW registers at boot"
+	depends on ARCH_SUPPORTS_ARCH_HW_INIT
+	help
+	  This option instructs Zephyr to force the initialization
+	  of architecture core registers during boot to their reset
+	  values. The option is useful when the Zephyr firmware
+	  image is chain-loaded by e.g. a bootloader and we need
+	  to guarantee that the architecture core HW registers
+	  are initialized to their warm reset values (as specified
```

Comment:
Actually no, @nvlsianpu (but this is a good question). Warm reset conditions is the minimum we need to satisfy, in general, so no need to set cold reset here. 

Might as well remove "warm" and leave just "reset" if this seems confusing :)

Reply:
Better to clarify what this kind of reset values means - having adjective her is good as this is certain kind of reset values

Reply:
IMHO description is a bit incomplete or even incorrect. Registers are just an interface to communicate with hardware. That said instead of setting some registers in its pristine state we want to get some HW internals back to its non-adultered state.

And an example is ARConnect IP (set of IP blocks to be used on SMP systems). There we have an interface of just 3 registers (one for command, one for data to write, one for data to read) for dealing with many very different blocks like cross-core IRQs, independent timer etc. And setting those 3 regs with just 1 set of values won't do the work.

Reply:
@abrodkin thanks
So you propose to reword "HW registers" to "Architecture core HW internals"? or something similar?

Reply:
Yep, to something like "Initialize internal architectural states at boot".

Reply:
Yeap, that's fine with me. Updated now @abrodkin.